### PR TITLE
zola: 0.17.2 -> 0.18.0

### PR DESCRIPTION
**Pull request by ivan770**
## Description of changes

Changes: https://github.com/getzola/zola/compare/v0.17.2...v0.18.0

This PR also removes `openssl` dependency, as it's unnecessary since version [0.10.0](https://github.com/getzola/zola/blob/master/CHANGELOG.md#other-5).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by 0x4A6F — APPROVED** [review]
Thanks and merry Christmas

**Review by alerque — APPROVED** [review]

**Comment by 0x4A6F** [comment]
@NixOS/darwin-maintainers Any suggestion how to fix the hash mismatch on darwin generally? There are other occurences in the rust ecosystem [e.g. atuin](https://github.com/NixOS/nixpkgs/pull/273427#discussion_r1422202343).

**Comment by viraptor** [comment]
I haven't seen this in rust yet, but it may be a similar thing to go's `proxyVendor` requirement for packages with problems with case sensitive names. I don't think rust has an equivalent at the moment... It may be easiest to have a hash for darwin and linux separately for now?

File: pkgs/by-name/zo/zola/package.nix [hunk 1]
```diff
@@ -1,51 +1,37 @@
...
   ];
+
   buildInputs = [
-    openssl
     oniguruma
-    libsass
   ] ++ lib.optionals stdenv.isDarwin [
-    CoreServices
+    darwin.apple_sdk.frameworks.CoreServices
   ];
```
Comment (viraptor) [inline pkgs/by-name/zo/zola/package.nix:35]:
```
  ] ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
    CoreServices SystemConfiguration
  ]);
```

**Review by alerque — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -36965,10 +36965,6 @@ with pkgs;
 
   zita-njbridge = callPackage ../applications/audio/zita-njbridge { };
 
-  zola = callPackage ../applications/misc/zola {
-    inherit (darwin.apple_sdk.frameworks) CoreServices;
-  };
-
```
Comment (camelpunch) [inline pkgs/top-level/all-packages.nix:36971]: Doesn't this prevent zola from being installable via standard methods?

Did you mean to only remove the inherit line?

Comment (wegank) [reply pkgs/top-level/all-packages.nix:36971]: It doesn't, see [this](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#name-based-package-directories):

> Packages found in the named-based structure do not need to be explicitly added to the `top-level/all-packages.nix` file unless they require overriding the default value of an implicit attribute.

Comment (camelpunch) [reply pkgs/top-level/all-packages.nix:36971]: Ah interesting! Thanks.

**Pull request merged**